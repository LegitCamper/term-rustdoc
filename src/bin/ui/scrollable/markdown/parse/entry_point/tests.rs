use super::{markdown_iter, parse};
use insta::{assert_debug_snapshot as snap, assert_display_snapshot as shot};

#[test]
fn parse_markdown() {
    let doc = r#"
# h1 `code`

aaa b *c* d **e**. ~xxx~ z.

1 *c **ss** d sadsad xxx* `yyyy`

```
let a = 1;
```

> rrr sss 
> tt

- [x] done!
    - nested list
- [ ] undone
    1. *a*
    2. `b`
"#;
    snap!(markdown_iter(doc).collect::<Vec<_>>());
    shot!(parse(doc), @r###"
    # h1 `code`

    aaa b c d e. xxx z.

    1 c ss d sadsad xxx `yyyy`

    let a = 1;

    rrr sss tt

    * [x] done!
      * nested list
    * [ ] undone
      1. a
      2. `b`

    "###);
}

#[test]
fn parse_markdown_links() {
    let doc = "
[a](b), [c], [e][c].

[c]: d

[`f`][c].

## h2 [c] [`h`][c]
";
    snap!(markdown_iter(doc).collect::<Vec<_>>());
    shot!(parse(doc), @r###"
    a, c, e.

    `f`.

    ## h2 c `h`

    "###);
}
