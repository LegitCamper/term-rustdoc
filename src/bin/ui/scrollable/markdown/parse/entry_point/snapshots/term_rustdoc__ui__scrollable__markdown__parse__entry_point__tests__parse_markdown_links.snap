---
source: src/bin/ui/scrollable/markdown/parse/entry_point/tests.rs
expression: "markdown_iter(doc).collect::<Vec<_>>()"
---
[
    (
        Start(
            Paragraph,
        ),
        1..22,
    ),
    (
        Start(
            Link {
                link_type: Inline,
                dest_url: Borrowed(
                    "b",
                ),
                title: Borrowed(
                    "",
                ),
                id: Borrowed(
                    "",
                ),
            },
        ),
        1..7,
    ),
    (
        Text(
            Borrowed(
                "a",
            ),
        ),
        2..3,
    ),
    (
        End(
            Link,
        ),
        1..7,
    ),
    (
        Text(
            Borrowed(
                ", ",
            ),
        ),
        7..9,
    ),
    (
        Start(
            Link {
                link_type: Shortcut,
                dest_url: Borrowed(
                    "d",
                ),
                title: Borrowed(
                    "",
                ),
                id: Borrowed(
                    "c",
                ),
            },
        ),
        9..12,
    ),
    (
        Text(
            Borrowed(
                "c",
            ),
        ),
        10..11,
    ),
    (
        End(
            Link,
        ),
        9..12,
    ),
    (
        Text(
            Borrowed(
                ", ",
            ),
        ),
        12..14,
    ),
    (
        Start(
            Link {
                link_type: Reference,
                dest_url: Borrowed(
                    "d",
                ),
                title: Borrowed(
                    "",
                ),
                id: Borrowed(
                    "c",
                ),
            },
        ),
        14..20,
    ),
    (
        Text(
            Borrowed(
                "e",
            ),
        ),
        15..16,
    ),
    (
        End(
            Link,
        ),
        14..20,
    ),
    (
        Text(
            Borrowed(
                ".",
            ),
        ),
        20..21,
    ),
    (
        End(
            Paragraph,
        ),
        1..22,
    ),
    (
        Start(
            Paragraph,
        ),
        31..41,
    ),
    (
        Start(
            Link {
                link_type: Reference,
                dest_url: Borrowed(
                    "d",
                ),
                title: Borrowed(
                    "",
                ),
                id: Borrowed(
                    "c",
                ),
            },
        ),
        31..39,
    ),
    (
        Code(
            Borrowed(
                "f",
            ),
        ),
        32..35,
    ),
    (
        End(
            Link,
        ),
        31..39,
    ),
    (
        Text(
            Borrowed(
                ".",
            ),
        ),
        39..40,
    ),
    (
        End(
            Paragraph,
        ),
        31..41,
    ),
    (
        Start(
            Heading {
                level: H2,
                id: None,
                classes: [],
                attrs: [],
            },
        ),
        42..61,
    ),
    (
        Text(
            Borrowed(
                "h2 ",
            ),
        ),
        45..48,
    ),
    (
        Start(
            Link {
                link_type: Shortcut,
                dest_url: Borrowed(
                    "d",
                ),
                title: Borrowed(
                    "",
                ),
                id: Borrowed(
                    "c",
                ),
            },
        ),
        48..51,
    ),
    (
        Text(
            Borrowed(
                "c",
            ),
        ),
        49..50,
    ),
    (
        End(
            Link,
        ),
        48..51,
    ),
    (
        Text(
            Borrowed(
                " ",
            ),
        ),
        51..52,
    ),
    (
        Start(
            Link {
                link_type: Reference,
                dest_url: Borrowed(
                    "d",
                ),
                title: Borrowed(
                    "",
                ),
                id: Borrowed(
                    "c",
                ),
            },
        ),
        52..60,
    ),
    (
        Code(
            Borrowed(
                "h",
            ),
        ),
        53..56,
    ),
    (
        End(
            Link,
        ),
        52..60,
    ),
    (
        End(
            Heading(
                H2,
            ),
        ),
        42..61,
    ),
    (
        Start(
            Paragraph,
        ),
        62..69,
    ),
    (
        Text(
            Borrowed(
                "m",
            ),
        ),
        62..63,
    ),
    (
        FootnoteReference(
            Borrowed(
                "n",
            ),
        ),
        63..67,
    ),
    (
        Text(
            Borrowed(
                ".",
            ),
        ),
        67..68,
    ),
    (
        End(
            Paragraph,
        ),
        62..69,
    ),
    (
        Start(
            FootnoteDefinition(
                Borrowed(
                    "n",
                ),
            ),
        ),
        70..81,
    ),
    (
        Start(
            Paragraph,
        ),
        76..81,
    ),
    (
        Text(
            Borrowed(
                "blah",
            ),
        ),
        76..80,
    ),
    (
        End(
            Paragraph,
        ),
        76..81,
    ),
    (
        End(
            FootnoteDefinition,
        ),
        70..81,
    ),
]
